MaxText/max_utils.py

```diff
@@ -229,8 +229,6 @@ def maybe_initialize_jax_distributed_system(raw_keys):
     max_logging.log("Jax distributed system initialized on CPUs!")
   elif (
       raw_keys["enable_checkpointing"]
-      and raw_keys["async_checkpointing"]
-      and raw_keys["compile_topology_num_slices"] == -1
```

Comment:
Shouldn't we keep and raw_keys["compile_topology_num_slices"] == -1? For xAOT we don't need to initialize the jax distributed system - we can compile e.g. on a CPU

Reply:
Ah I actually added this a while ago with GPU AoT: https://github.com/google/maxtext/blob/116c3f2a918a78f037b17657b75c8ed5268a408e/MaxText/max_utils.py#L215-L217

So it's a redundant check now.